### PR TITLE
SQL, Tests: FOR SYSTEM_TIME for VIEWs [closes #98]

### LINKED ISSUE #98: 0.7: "AS OF" query for VIEW
# Problem
You cannot perform an "As Of" query through a view definition based on a Versioned table, you get the following error

```
ERROR 4031 (HY000): System Versioning required: `FOR SYSTEM_TIME` query
```

# Solution
A view can be a restricted projection of data, joins etc. on the underlying tables & views. If you add a predicate to a query with a view, then this is simply added to the overall query predicate. It would seem that "As Of" would behave in the same way, a further restriction to the underlying query.

The spec does not seem to call out any explicit behavior for views.

# Reproduce
```
create or replace table dept (
  dept_id int(10) primary key, 
  name varchar(100)
)
with system versioning
;

insert into dept (dept_id, name) values (10, "sales");

select vtq_commit_ts(sys_trx_start) into @ts_1 from dept where dept_id=10;

create view dept_view as select * from dept where name = "sales";

select * from dept_view;

select * from dept for system_time as of timestamp @ts_1;

select * from dept_view where dept_id = 10;
+---------+-------+
| dept_id | name  |
+---------+-------+
|      10 | sales |
+---------+-------+

select * from dept_view for system_time as of timestamp @ts_1;
ERROR 4031 (HY000): System Versioning required: `FOR SYSTEM_TIME` query
```

### PR DESCRIPTION
**Pull request by kevgs**

### CODE REVIEW

File: sql/sql_view.cc [hunk 1]
```diff
@@ -453,6 +453,17 @@ bool mysql_create_view(THD *thd, TABLE_LIST *views,
     goto err;
   }
 
+  /* Implicitly add versioning fields in needed */
```
Comment (midenok) [inline sql/sql_view.cc]: Typo? *if needed*

File: sql/sql_select.cc [hunk 2]
```diff
@@ -684,6 +684,13 @@ int vers_setup_select(THD *thd, TABLE_LIST *tables, COND **where_expr,
     DBUG_RETURN(0);
   }
 
+  vers_select_conds_t *vers_conds_for_view= NULL;
```
Comment (midenok) [inline sql/sql_select.cc]: Local variable, no need for such long name. Use `view_conds` instead.

File: sql/sql_select.cc [hunk 3]
```diff
@@ -16499,6 +16509,7 @@ create_tmp_table(THD *thd, TMP_TABLE_PARAM *param, List<Item> &fields,
   }
 
   field_count=param->field_count+param->func_count+param->sum_func_count;
+
```
Comment (midenok) [inline sql/sql_select.cc]: Junk hunk.

**Review by midenok — APPROVED** [review]
Please, fix minor notes.

**Comment by kevgs** [comment]
Please, review.

File: sql/sql_select.cc [hunk 4]
```diff
@@ -16705,6 +16707,20 @@ create_tmp_table(THD *thd, TMP_TABLE_PARAM *param, List<Item> &fields,
...
+      {
+        Item_field *item_field= (Item_field *)item;
+        Field *field= item_field->field;
+        TABLE_SHARE *s= field->table->s;
+        if (s->versioned)
+        {
+          if (field->flags & VERS_SYS_START_FLAG && fieldnr == field_count - 2)
+            sys_trx_start= new_field;
+          if (field->flags & VERS_SYS_END_FLAG && fieldnr == field_count - 1)
+            sys_trx_end= new_field;
```
Comment (midenok) [inline sql/sql_select.cc]: `sys_trx_start` and `sys_trx_end` can't be same: use `else`.

**Review by midenok — APPROVED** [review]
Minor comment.

**Pull request merged**